lib/plugins/slack/attachements.js

```diff
@@ -12,89 +13,100 @@ function getMetric(metric) {
         },
         fullyLoaded: {
-          name: 'Fully loaded',
-          metric: base.browsertime ? base.browsertime.pageSummary.statistics.timings.fullyLoaded : undefined
+          name: 'Fully Loaded',
+          metric: get(base.browsertime, 'pageSummary.statistics.timings.fullyLoaded')
         },
         domContentLoadedTime: {
           name: 'domContentLoadedTime',
-          metric: base.browsertime ? base.browsertime.pageSummary.statistics.timings.pageTimings.domContentLoadedTime : undefined
+          metric: get(base.browsertime, 'pageSummary.statistics.timings.pageTimings.domContentLoadedTime')
         },
         rumSpeedIndex: {
           name: 'RUM Speed Index',
-          metric: base.browsertime ? base.browsertime.pageSummary.statistics.timings.rumSpeedIndex : undefined
+          metric: get(base.browsertime, 'pageSummary.statistics.timings.rumSpeedIndex')
         },
         coachScore: {
           name: 'Coach score',
-          metric: base.coach ? base.coach.pageSummary.advice.performance.score : undefined
+          metric: get(base.coach, 'pageSummary.advice.performance.score')
         },
         transferSize: {
           name: 'Page transfer size',
-          metric: base.pagexray ? h.size.format(base.pagexray.pageSummary.transferSize) : undefined
+          metric: h.size.format(get(base.pagexray, 'pageSummary.transferSize'))
         },
         transferRequests: {
           name: 'Requests',
-          metric: base.pagexray ? base.pagexray.pageSummary.requests : undefined
+          metric: get(base.pagexray, 'pageSummary.requests')
         }
-      }
+      };
 
       const fields = [];
       for (const key of Object.keys(metrics)) {
-        if (metrics[key] && metrics[key].metric !== undefined) {
+        const metric = metrics[key];
+        if (metric.metric !== undefined) {
           fields.push({
-            title: metrics[key].name,
-            value: getMetric(metrics[key].metric),
+            title: metric.name,
+            value: getMetric(metric.metric),
             short: true
           })
         }
       }
 
       // add all the errors
-      if (dataCollection.urlPages[url].errors) {
-        for (const key of Object.keys(dataCollection.urlPages[url].errors)) {
+      if (results.errors) {
+        for (const key of Object.keys(results.errors)) {
           fields.push({
             title: key + ' error',
-            value: dataCollection.urlPages[url].errors[key],
+            value: results.errors[key],
             short: false
           })
         }
       }
 
-      let color = metrics[options.limitMetric] ? metrics[options.limitMetric].metric > options.limitError ? 'good' : metrics[options.limitMetric].metric > options.limitWarning ? 'warning' : 'danger' : 'danger';
+      let color = 'danger';
 
-      attachements.push({
+      const limitMetric = metrics[options.limitMetric];
```

Comment:
This looks strange to me. Shouldn't 'good' be the default? Also, the other checks seems backwards. For coachScore, higher is better, but for speedIndex and firstVisualChange the reverse is true.